LEAF_Request_Portal/form.php

```diff
@@ -3048,14 +3053,32 @@ public function query($inQuery)
                         AND deleted = 0', array());
             foreach ($res2 as $item)
             {
-                if($data[$item['recordID']]['recordResolutionData']['fulfillmentTime'] == null
-                    || $data[$item['recordID']]['recordResolutionData']['fulfillmentTime'] < $item['fulfillmentTime']) {
+                if($data[$item['recordID']]['recordResolutionData']['fulfillmentTime'] == null || $data[$item['recordID']]['recordResolutionData']['fulfillmentTime'] < $item['fulfillmentTime']) {
                     $data[$item['recordID']]['recordResolutionData']['lastStatus'] = $item['lastStatus'];
                     $data[$item['recordID']]['recordResolutionData']['fulfillmentTime'] = $item['fulfillmentTime'];
                 }
             }
         }
 
+        if($joinRecordResolutionBy) {
```

Comment:
Ambiguous "If".  This needs to search for a specific state.  Is it "TRUE"; does it exist; is it not an empty string?  PHP is great but since it allows you to fill any variable with anything, when you test it you need to specify exactly what you're looking for in that test.

Reply:
This is set as false by default on line 2851 of form.tpl:         $joinRecordResolutionBy = false;
and then is triggered by the checkbox request, line 2892 of form.tpl: case 'recordResolutionBy':
                        $joinRecordResolutionBy = true;